### PR TITLE
refactoring in preparation for multiplayer launcher integration

### PR DESCRIPTION
**Pull request by rfomin**
It's not finished, I just decided to rename the Source dir to src and I want to know your opinion on this. :smile:

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
I agree that `src` is the most natural choice for the source directory, but I am afraid changing the name now breaks some established workflow. 

Immediately after changing all line endings to Unix style on the original source releases, I created the "upstream" tag. So, I can always compare the current source state with the original WinMBF release by running e.g. `git diff upstream -- Source/wi_stuff.c` and similar. I guess this won't be possible anymore when renanimg the entire directory.

Also, I guess that features like `git blame` would always reference the renaming step as the last change to a source file and would thus get rendered meaningless.

File: Source/d_main.c [hunk 1]
```diff
@@ -2303,6 +2246,11 @@ void D_DoomMain(void)
       debugfile = fopen(filename,"w");
     }
 
+  if (!demorecording)
+    I_AtExit(D_EndDoom, false);
+
+  TryRunTics();
```
Comment (fabiangreffrath) [inline Source/d_main.c:2252]: Why TryRunTics here?

Comment (rfomin) [reply Source/d_main.c:2252]: This is from Choco, but it does not apply to this PR, I staged it by accident.

**Comment by rfomin** [comment]
> Immediately after changing all line endings to Unix style on the original source releases, I created the "upstream" tag. So, I can always compare the current source state with the original WinMBF release by running e.g. `git diff upstream -- Source/wi_stuff.c` and similar. I guess this won't be possible anymore when renanimg the entire directory.

There is no git magic to somehow edit the entire history, right? Eh.

File: Source/d_main.c [hunk 2]
```diff
@@ -2303,6 +2246,9 @@ void D_DoomMain(void)
       debugfile = fopen(filename,"w");
     }
 
+  if (!demorecording)
+    I_AtExit(D_EndDoom, false);
```
Comment (fabiangreffrath) [inline Source/d_main.c:2250]: Brackets or not?

File: Source/m_menu.c [hunk 3]
```diff
@@ -5198,10 +5199,11 @@ boolean M_Responder (event_t* ev)
 
         if (M_InputActivated(input_demo_fforward))
         {
-          if (demoplayback && singledemo && !demo_skipping)
+          if (demoplayback && singledemo && !demo_skipping && !fastdemo)
           {
-            fastdemo = !fastdemo;
-            I_SetFastdemoTimer();
+            static boolean fastdemo_timer = false;
```
Comment (fabiangreffrath) [inline Source/m_menu.c:5204]: SO , it's not possible to turn a fastdemo into a normal demo anymore?

File: Source/i_main.c [hunk 4]
```diff
@@ -77,6 +104,14 @@ static void I_Signal(void)
     }
 }
 
+//
+// D_DoomMain()
+// Not a globally visible function, just included for source reference,
+// calls all startup code, parses command line options.
+//
+
+void D_DoomMain(void);
```
Comment (fabiangreffrath) [inline Source/i_main.c:113]: Why is this here?

File: Source/m_menu.c (hunk 3 above)
Comment (rfomin) [reply Source/m_menu.c:5204]: The `fastdemo` variable indicates the -fastdemo command line parameter. It is possible to enable the fastdemo timer during normal demo playback, but not with the -fastdemo parameter as it counts FPS.

Comment (fabiangreffrath) [reply Source/m_menu.c:5204]: Alright! 👍

File: Source/i_main.c (hunk 4 above)
Comment (rfomin) [reply Source/i_main.c:113]: I removed the `d_main.h` header. The idea is to use `i_main.c` in the setup and in the main game like in Choco.

Comment (fabiangreffrath) [reply Source/i_main.c:113]: Ack. 👍

File: Source/d_iwad.c [hunk 5]
```diff
@@ -18,12 +18,32 @@
...
 #include "d_iwad.h"
 #include "m_argv.h"
 #include "m_misc2.h"
 
+static const iwad_t iwads[] =
+{
+    { "doom2.wad",     doom2,      commercial, "Doom II" },
+    { "plutonia.wad",  pack_plut,  commercial, "Final Doom: Plutonia Experiment" },
+    { "tnt.wad",       pack_tnt,   commercial, "Final Doom: TNT: Evilution" },
+    { "doom.wad",      doom,       retail,     "Doom" },
```
Comment (fabiangreffrath) [inline Source/d_iwad.c:32]: Do we have any chance to detect non-retail, i.e. non-Ultimate Doom 1?

**Review by fabiangreffrath — APPROVED** [review]

File: Source/d_iwad.c (hunk 5 above)
Comment (rfomin) [reply Source/d_iwad.c:32]: It seems not. So we still need to check the contents of the IWAD if `filename == "doom.wad"`.

File: Source/d_iwad.c [hunk 6]
```diff
@@ -689,6 +689,14 @@ char *D_FindIWADFile(GameMode_t *mode, GameMission_t *mission)
         int i;
         const char *name = M_BaseName(result);
 
+        // "doom.wad" may be retail or registered
+        if (!strcasecmp("doom.wad", name))
```
[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply Source/d_iwad.c:693]: ... with a comment that we cannot know if "registered" or "retail" from the file name.

[the remark this answers is not in the document]
Comment (rfomin) [reply Source/d_iwad.c]: Done in https://github.com/fabiangreffrath/woof/pull/572/commits/e29a8b1007aa63cb9c0cd84f0dfa9e32a64260b6

**Comment by fabiangreffrath** [comment]
While at it, could you please fix the "CheckIWAD: IWAD tag %s not present\n" warning message in the now called `CheckIWAD()`? It's confusing to have the file name in the middle of the warning.

File: Source/d_iwad.c [hunk 7]
```diff
@@ -18,12 +18,33 @@
...
 #include "m_misc2.h"
 
+static const iwad_t iwads[] =
+{
+    { "doom2.wad",     doom2,      commercial, "Doom II" },
+    { "plutonia.wad",  pack_plut,  commercial, "Final Doom: Plutonia Experiment" },
+    { "tnt.wad",       pack_tnt,   commercial, "Final Doom: TNT: Evilution" },
+    // "doom.wad" may be retail or registered
+    { "doom.wad",      doom,       indetermined, "Doom" },
+    { "doom1.wad",     doom,       shareware,  "Doom Shareware" },
```
[the remark this answers is not in the document]
Comment (rfomin) [reply Source/d_iwad.c:34]: To be honest, I care a lot :smile:  And about formatting in general. I'll fix it.

**Comment by fabiangreffrath** [comment]
> I guess this won't be possible anymore when renanimg the entire directory.

Actually it's still possible using the `upstream` tag and more concrete file names:
`git diff upstream:Source/hu_stuff.c src/hu_stuff.c`

So, maybe we should just do it. Better now than never...

**Comment by rfomin** [comment]
> Actually it's still possible using the `upstream` tag and more concrete file names: `git diff upstream:Source/hu_stuff.c src/hu_stuff.c`
> 
> So, maybe we should just do it. Better now than never...

If it's ok for you, I'm all for it :+1:

**Comment by rfomin** [comment]
> Could you please revive this part of the PR then?

Cherry pick it? I just used this command: `git mv Source src`.
